flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -110,6 +110,14 @@
 			" resolved through the parent ClassLoader first. A pattern is a simple prefix that is checked against" +
 			" the fully qualified class name. These patterns are appended to \"" + ALWAYS_PARENT_FIRST_LOADER_PATTERNS.key() + "\".");
 
+	@Documentation.Section(Documentation.Sections.EXPERT_CLASS_LOADING)
+	public static final ConfigOption<Boolean> FAIL_USER_CLASS_LOADING_METASPACE_OOM = ConfigOptions
+		.key("classloader.fail.on.metaspace.oom.error")
+		.booleanType()
+		.defaultValue(true)
+		.withDescription("Sets whether to fail and exit TaskManager JVM process if 'OutOfMemoryError: Metaspace' is " +
+			"thrown trying to load a user code class.");
```

Comment:
```suggestion
			"thrown while trying to load a user code class.");
```

Reply:
I would remove the reference to `user code` in the description and field name since this change also applies to plugins.

Reply:
I would add then `user or plugin` because if we drop `user` at all, it will be confusing. The option is really a fallback only for user scope class loading, not for all places where this exception can be thrown.

I would keep the internal variable name as it is because this is how we call the respective classes it tweaks at the moment: `FlinkUserCodeClassLoader`.

Reply:
Nevermind, turns out the plugin system has its own custom class loader, and is thus unaffected by this change.